Assets/MirageXR/ContentTypes/Video/Editor/Mobile/CaptionGenerator.cs

```diff
@@ -0,0 +1,300 @@
+#pragma warning disable 0649
+
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+using UnityEngine.UI;
+using IBM.Watson.SpeechToText.V1;
+using IBM.Cloud.SDK;
+using IBM.Cloud.SDK.Authentication;
+using IBM.Cloud.SDK.Authentication.Iam;
+using IBM.Cloud.SDK.Utilities;
+using IBM.Cloud.SDK.DataTypes;
+using TMPro;
+
+namespace IBM.Watsson.Examples
+{
+    public class CaptionGenerator : MonoBehaviour
+    {
+        #region PLEASE SET THESE VARIABLES IN THE INSPECTOR
+
+        private string _serviceUrl;
+        [Tooltip("Text field to display the results of streaming.")]
+        [SerializeField]
+        private TMP_Text ResultsField;
+        private string _iamApikey;
+        private string _recognizeModel;
+
+        #endregion
+
+        private int _recordingRoutine = 0;
+        private string _microphoneID = null;
+        private AudioClip _recording = null;
+        private int _recordingBufferSize = 1;
+        private int _recordingHZ = 22050;
+        private SpeechToTextService _service;
+        private string accumulatedText = "";
+        [SerializeField] private Button _stopButton;
```

Comment:
remove region comments and sort the fields: first private fields with the [SerializeField] attribute, then just private fields. write the attribute on the same line as the field.

Reply:
also follow the formatting throughout the file